beacon_node/network/src/sync/block_lookups/mod.rs

```diff
@@ -248,22 +252,48 @@ impl<T: BeaconChainTypes> BlockLookups<T> {
                 // blocks on top of A forming A -> C. The malicious peer forces us to fetch C
                 // from it, which will result in parent A hitting the chain_too_long error. Then
                 // the valid chain A -> B is dropped too.
-                if let Ok(block_to_drop) = find_oldest_fork_ancestor(parent_chains, chain_idx) {
-                    // Drop all lookups descending from the child of the too long parent chain
-                    if let Some((lookup_id, lookup)) = self
+                //
+                // `find_oldest_fork_ancestor` should never return Err, unwrapping to tip for
+                // complete-ness
+                let parent_chain_tip = parent_chain.tip;
+                let block_to_drop =
+                    find_oldest_fork_ancestor(parent_chains, chain_idx).unwrap_or(parent_chain_tip);
+                // Drop all lookups descending from the child of the too long parent chain
+                if let Some((lookup_id, lookup)) = self
+                    .single_block_lookups
+                    .iter()
+                    .find(|(_, l)| l.block_root() == block_to_drop)
+                {
+                    // If a lookup chain is too long we can't distinguish a valid chain between a
+                    // malicious chain. We must attempt to sync this chain to not lose liveness. If
+                    // the chain grows too long we stop lookup sync and transition this head to
+                    // forwards range sync. We need to tell range sync what head to sync too and
+                    // from which peers. The lookup of the very tip of this chain may contain zero
+                    // peers if it's the parent child lookup. So we do a bit a trick here:
+                    // - Tell range sync to sync to the tip's root (if available, else its ancestor)
+                    // - But use all peers in the ancestor lookup, which should have at least one
+                    //   peer and its peer set is a strict superset of the tip's lookup.
```

Comment:
```suggestion
                    // If a lookup chain is too long, we can't distinguish a valid chain from a 
                    // malicious one. We must attempt to sync this chain to not lose liveness. If 
                    // the chain grows too long, we stop lookup sync and transition this head to 
                    // forward range sync. We need to tell range sync which head to sync to, and 
                    // from which peers. The lookup of the very tip of this chain may contain zero 
                    // peers if it's the parent-child lookup. So we do a bit of a trick here:
                    // - Tell range sync to sync to the tip's root (if available, else its ancestor)
                    // - But use all peers in the ancestor lookup, which should have at least one 
                    //   peer, and its peer set is a strict superset of the tip's lookup.
```

Minor typo nit that reads clearer for me